WebService/WebService/ErrorHandling/ExceptionExtension.cs

```diff
@@ -18,18 +18,18 @@ public static void ConfigureExceptionHandler(this IApplicationBuilder app, ILogg
             {
                 appError.Run(async context =>
                 {
-                    context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
                     context.Response.ContentType = "application/json";
+                    context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
```

Comment:
Is InternalServerError the only possible error in backend?

Reply:
Yes and no. The way it's set up is that the error handler would catch any errors or exceptions in backend and return 500 to the user to let them know, that it's our fault. So whatever breaks on our side, the user would get 500 and there would be an error logged in the logfile with details of when and why the error happened. There are some other possible errors, but they are handled in the methods that can produce those errors. For example, if you look at DeleteBar method in BarsController.cs, there's a NotFound() returned, because we can check if we actually found the bar that was given to us and if we don't have it, we tell the user, that it's not there. It's not an error, however, just a way to tell the user what happened.

Reply:
So the 500 should be caught and returned only if it's directly the fault of our code and every other scenario still has to be checked according to what method we are implementing, if that method can give the user more information than just that it was an internal error.
For reference: https://restfulapi.net/http-status-codes/